packages/nx/src/generators/utils/project-configuration.spec.ts

```diff
@@ -207,4 +208,25 @@ describe('project configuration', () => {
       });
     });
   });
+
+  describe('getProjectJsonProjects', () => {
+    it('should not find package.json based projects', () => {
```

Comment:
`package.json` projects are still editable.  Why do you want to only list `project.json` projects?

Reply:
`updateProjectConfiguration` will throw on package.json based projects. This means something like this would throw:

```ts
for (const [project, config] of getProjects(tree).entries()) {
  updateProjectConfiguration(project, modifyConfig(config))
}
```

Its a common foot gun that we've messed up ourselves multiple times.